velox/core/QueryConfig.h

```diff
@@ -97,10 +97,15 @@ class QueryConfig {
   static constexpr const char* kMaxPartitionedOutputBufferSize =
       "driver.max-page-partitioning-buffer-size";
 
-  /// Preffered number of rows to be returned by operators from
+  /// Preferred size of batches in bytes to be returned by operators from
   /// Operator::getOutput.
-  static constexpr const char* kPreferredOutputBatchSize =
-      "preferred_output_batch_size";
+  static constexpr const char* kPreferredOutputBatchBytes =
+      "preferred_output_batch_bytes";
+
+  /// Preferred number of rows to be returned by operators from
```

Comment:
The comment says "preferred", but the property name was changed to "default". It is actually confusing to use the name "default" because that property itself has a default value of 1024. Perhaps, keep the original name, but explain how the two properties are being used.

preferred_output_batch_bytes is used when operator can estimate row sizes.
preferred_output_batch_rows is used when operator cannot estimate row sizes.

Reply:
Revised